.github/workflows/dotnet.yml

```
@@ -0,0 +1,39 @@
+name: Chorus .NET6
+
+on:
+  [push, pull_request, workflow_dispatch]
+
+jobs:
+  build_test_deploy:
+    env:
+      NUGET_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+    permissions: write-all
+    
+    steps:
+    - uses: actions/checkout@v3
```

From a security POV it would be better to specify the commit SHA instead of the tag - a malicious user/repo owner could change the tag but he can't change the sha.

```suggestion
    - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
```